integreat_cms/static/src/js/analytics/hix-widget.ts

```diff
@@ -167,14 +168,33 @@ window.addEventListener("load", async () => {
     document.getElementById("btn-update-hix-value").addEventListener("click", async (event) => {
         document.getElementById("hix-loading")?.classList.remove("hidden");
         event.preventDefault();
-        const newHixValue = await getHixValue((event.target as HTMLElement).dataset.url);
+        const newHixValue = await getHixValue();
         updateChart(chart, newHixValue);
         toggleMTAvailability(newHixValue);
     });
 
+    const initHixValue = async () => {
+        if (!getContent().trim()) {
+            setHixLabelState("no-content");
+            return;
+        }
+
+        const initialHixValue =
+            parseFloat(document.getElementById("hix-container").dataset.initialHixScore) || (await getHixValue());
+
+        setHixLabelState("updated");
+        updateChart(chart, initialHixValue);
+        toggleMTAvailability(initialHixValue);
+    };
+
     // Set listener, that checks, if tinyMCE content has changed to update the
     // HIX value status
     document.querySelectorAll("[data-content-changed]").forEach((element) => {
+        // make sure initHixValue is called only after tinyMCE is initialized
+        element.addEventListener("tinyMCEInitialized", () => {
+            initHixValue();
```

Comment:
```suggestion
        element.addEventListener("tinyMCEInitialized", async () => {
            await initHixValue();
```
I believe this has to be async, otherwise the loading overlay is removed instantly, without waiting for the result.